sklearn/ensemble/_stacking.py

```diff
@@ -0,0 +1,555 @@
+
+    passthrough : bool, optional
+        Whether or not to concatenate the original data ``X`` with the output
+        of ``estimators`` to feed the ``final_estimator``. The default is
+        False.
+
+    n_jobs : int, optional (default=1)
+        The number of jobs to ``fit`` the ``estimators`` in parallel. If
+        -1, then the number of jobs is set to the number of cores.
+
+    random_state : int, RandomState instance or None, optional (default=None)
+        If int, random_state is the seed used by the random number generator;
+        If RandomState instance, random_state is the random number generator;
+        If None, the random number generator is the RandomState instance used
+        by `np.random`. Used to set the ``cv``.
+
+    Attributes
+    ----------
+    estimators_ : list of estimator object
+        The base estimators fitted.
+
+    final_estimator_ : estimator object
+        The regressor to stacked the base estimators fitted.
+
+    predict_method_ : list of string
+        The method used by each base estimator.
+
+    References
+    ----------
+    .. [1] Wolpert, David H. "Stacked generalization." Neural networks 5.2
+       (1992): 241-259.
+
+    Examples
+    --------
+    >>> from sklearn.datasets import load_diabetes
+    >>> X, y = load_diabetes(return_X_y=True)
+    >>> from sklearn.linear_model import LinearRegression
+    >>> from sklearn.svm import LinearSVR
+    >>> from sklearn.ensemble import RandomForestRegressor
+    >>> from sklearn.ensemble import StackingRegressor
+    >>> estimators = [
+    ...     ('lr', LinearRegression()),
+    ...     ('svr', LinearSVR(tol=1e-1, random_state=42))
+    ... ]
+    >>> reg = StackingRegressor(
+    ...     estimators=estimators,
+    ...     final_estimator=RandomForestRegressor(n_estimators=10,
+    ...                                           random_state=42),
+    ...     cv=5
+    ... )
+    >>> from sklearn.model_selection import train_test_split
+    >>> X_train, X_test, y_train, y_test = train_test_split(
+    ... X, y, random_state=42
+    ... )
+    >>> reg.fit(X_train, y_train).score(X_test, y_test) # doctest: +ELLIPSIS
+    0...
+
+    """
+    def __init__(self, estimators, final_estimator=None, cv=None,
+                 predict_method='auto', passthrough=False, n_jobs=1,
+                 random_state=None, verbose=0):
+        super().__init__(
+            estimators=estimators,
+            final_estimator=final_estimator,
+            cv=cv,
+            predict_method=predict_method,
+            passthrough=passthrough,
+            n_jobs=n_jobs,
+            random_state=random_state,
+            verbose=verbose
+        )
+
+    def _validate_meta_estimator(self):
+        super()._validate_final_estimator(
+            default=LinearRegression()
+        )
+        if not is_regressor(self.final_estimator_):
+            raise AttributeError(
+                "'final_estimator' attribute should be a regressor. Got {!r}"
```

Comment:
about {!r}: if it's a big pipeline the error message can potentially be very long and hard to read. I think we just print the class name in these cases? (which isn't helpful with pipelines anyway, I agree.)